src/services/newMessagesService.ts

```diff
@@ -541,11 +552,54 @@ export default class NewMessagesService {
                   return ResponseErrorStatusNotDefinedInSpec(response);
               }
             })
+          ),
+          TE.chainW((response) =>
+            this.validateThirdPartyMessageResponse(message, response)
           )
         )
       )
     );
 
+  private readonly validateThirdPartyMessageResponse = (
+    message: MessageWithThirdPartyData,
+    response: ThirdPartyMessageDetails
+  ): TE.TaskEither<IResponseErrorBadGateway, ThirdPartyMessage> => {
+    if (message.sender_service_id === PN_SERVICE_ID) {
```

Comment:
may we have a comment about this conditional statement here, expanding why PN is threated differently.

Reply:
solved here [added comment for PN check in third party validation](https://github.com/pagopa/io-backend/pull/1079/commits/057af95cd30ea6d43003c3d774323a0d84326864)